bouncer/shared/try_runtime_upgrade.ts

```diff
@@ -2,32 +2,53 @@
 // https://github.com/paritytech/try-runtime-cli
 
 import { ApiPromise } from '@polkadot/api';
-import { execSync } from 'child_process';
 import path from 'path';
-import os from 'os';
-import fs from 'fs';
 import { compileBinaries } from './utils/compile_binaries';
+import { createTmpDirIfNotExists, execWithRustLog } from './utils/exec_with_log';
 
-function tryRuntimeCommand(runtimePath: string, blockParam: string, networkUrl: string) {
-  // Create a temporary file for capturing stderr
-  const stderrFile = path.join(os.tmpdir(), `cmd-stderr-${Date.now()}`);
-  try {
-    execSync(
-      // TODO: Replace pre-and-post with all after the SDK issue paritytech/polkadot-sdk#2560 is merged.
-      `try-runtime --runtime ${runtimePath} on-runtime-upgrade --disable-spec-version-check --disable-idempotency-checks --checks pre-and-post ${blockParam} --uri ${networkUrl} 2> ${stderrFile}`,
-      { env: { ...process.env, RUST_LOG: 'runtime::executive=debug' } },
-    );
-    console.log(`try-runtime success for blockParam ${blockParam}`);
-  } catch (e) {
-    console.error(`try-runtime failed for blockParam ${blockParam}`);
-    const stderrOutput = fs.readFileSync(stderrFile, 'utf8');
-    console.error(e);
-    console.error('Command failed: Command output:', stderrOutput);
+function createSnapshotFile(networkUrl: string, blockHash: string) {
+  const blockParam = blockHash === 'latest' ? '' : `--at ${blockHash}`;
+  const snapshotFolder = createTmpDirIfNotExists('chainflip/snapshots/');
+  const snapshotOutputPath = path.join(snapshotFolder, `snapshot-at-${blockHash}.snap`);
 
-    fs.unlinkSync(stderrFile);
+  console.log('Writing snapshot to: ', snapshotOutputPath);
 
-    process.exit(-1);
+  execWithRustLog(
+    `try-runtime create-snapshot ${blockParam} --uri ${networkUrl} ${snapshotOutputPath}`,
+    `create-snapshot-${blockHash}`,
+    'runtime::executive=debug',
+    (success) => {
+      if (!success) {
+        console.error('Failed to create snapshot.');
+      }
+      process.exitCode = 1;
+    },
+  );
+}
+
+function tryRuntimeCommand(runtimePath: string, blockHash: 'latest' | string, networkUrl: string) {
```

Comment:
This is just refactored to use `execWithRustLog` and write a snapshot if it fails.